Src/Particle/AMReX_StructOfArrays.H

```diff
@@ -19,11 +23,18 @@ struct StructOfArrays {
     using RealVector = amrex::PODVector<ParticleReal, Allocator<ParticleReal> >;
     using IntVector = amrex::PODVector<int, Allocator<int> >;
 
-    void define (int a_num_runtime_real, int a_num_runtime_int)
+    void define (
+        int a_num_runtime_real,
+        int a_num_runtime_int,
+        std::vector<std::string>* soa_rdata_names=nullptr,
+        std::vector<std::string>* soa_idata_names=nullptr
```

Comment:
Yes, this is indeed not super save... Should we store a reference to a PC instead?

Or store the names in `StructOfArrays` and look them up when we need them from PC and ParticleTile?